src/log/file_appender.cpp

```diff
@@ -67,9 +63,22 @@ namespace fc {
 
          void rotate_files( bool initializing = false )
          {
-             FC_ASSERT( cfg.rotate );
+             if( !cfg.rotate ) return;
+
              fc::time_point now = time_point::now();
-             fc::time_point_sec start_time = get_file_start_time( now, cfg.rotation_interval );
+             if( now < _next_file_time ) return;
+
+             int64_t new_file_number = now.sec_since_epoch() / _interval_seconds;
+             if( initializing )
+                _current_file_number.store( new_file_number );
```

Comment:
If `_next_file_time` is `atomic`, we can build logic around it to replace `_current_file_number`, to save one member variable. ;-) The current code would work too though.